common/src/main/java/io/crate/types/TimestampType.java

```diff
@@ -29,44 +35,115 @@
 import java.time.format.ResolverStyle;
 import java.time.temporal.TemporalAccessor;
 import java.util.Locale;
+import java.util.function.Function;
 
 import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE;
 import static java.time.format.DateTimeFormatter.ISO_LOCAL_TIME;
 
-public final class TimestampType extends BaseTimestampType {
+public final class TimestampType extends DataType<Long>
+    implements FixedWidthType, Streamer<Long> {
 
-    public static final TimestampType INSTANCE = new TimestampType();
-    public static final int ID = 15;
+    public static final int ID_WITH_TZ = 15;
+    public static final int ID_WITHOUT_TZ = 11;
 
-    private static final DateTimeFormatter PARSER = new DateTimeFormatterBuilder()
-        .parseCaseInsensitive()
-        .append(ISO_LOCAL_DATE)
-        .optionalStart()
-            .appendLiteral('T')
-            .append(ISO_LOCAL_TIME)
-            .optionalStart()
-                .appendPattern("[VV][x][xx][xxx]")
-        .toFormatter(Locale.ENGLISH).withResolverStyle(ResolverStyle.STRICT);
+    public static final TimestampType WITH_TZ = new TimestampType(
```

Comment:
I think I would also add the `INSTANCE` here (and below) to make it explicit that they're actually timestamp instances.
It seems more clear when using them.

What do you think?